algorithms/score_plot.py

```diff
@@ -1,49 +1,106 @@
 import matplotlib.pyplot as plt
+from typing import Type, Optional
 
-from dataloader.base_recording import BaseRecording
 from dataloader.syscall import Syscall
+from dataloader.base_recording import BaseRecording
+
 
 # adjusting plot parameters
-plt.rcParams.update({"font.size": 26,
+plt.rcParams.update({"font.size": 20,
                      "figure.figsize": (55, 40),
-                     "agg.path.chunksize": 10000})
+                     "agg.path.chunksize": 10000,
+                     "legend.fontsize": 10,
+                     "axes.titlesize": 15,
+                     "axes.labelsize": 12,
+                     "xtick.labelsize": 10,
+                     "ytick.labelsize": 10},)
 
 
-class ScorePlot:
+THRESHOLD_COLOR_LIST = ["teal",
+                        "purple",
+                        "darkolivegreen",
+                        "darkblue",
+                        "maroon"]
 
-    def __init__(self, scenario_path):
+PLOT_COLOR_LIST = ["cadetblue",
+                   "mediumorchid",
+                   "olivedrab",
+                   "cornflowerblue",
+                   "tomato"]
 
+EXPLOIT_COLOR = "lightgray"
+
+
+class ScorePlot:
+    """
+    Plot anomaly scores while testing.
+    Needs to be initialized in main handed over as parameter to IDS.
+    Args:
+    building_blocks(list): list of bbs of which threshold and anomaly scores
+                           are being extracted. !!! needs to be decider with threshold!!!
+    scenario_path(str): scenario path for title of plot.
+    filename(str): filename if plot should be persisted.
+    Attributes:
+    _scenario_path (str):
+    """
+
+    def __init__(self,
+                 building_blocks: list,
+                 scenario_path: str,
+                 filename: Optional[str] = None):
         self._scenario_path = scenario_path
+        self._bb_list = building_blocks
         self._figure = None
-        self._anomaly_scores_exploits = []
-        self._anomaly_scores_no_exploits = []
+        self._filename = filename
+        self._anomaly_scores_exploits = {}
+        self._anomaly_scores_no_exploits = {}
         self._first_syscall_of_exploit_recording_index_list = []
         self._first_syscall_of_normal_recording_index_list = []
         self._first_syscall_after_exploit_index_list = []
         self._exploit_time = None
         self._first_sys_after_exploit = False
-        self.threshold = 0.0
+        self._thresholds = {}
         self._first_syscall_of_cfp_list_exploit = []
         self._last_syscall_of_cfp_list_exploit = []
         self._first_syscall_of_cfp_list_normal = []
         self._last_syscall_of_cfp_list_normal = []
 
-    def new_recording(self, recording: BaseRecording):
+    def set_threshold(self):
+        """
+        get threshold value from all building blocks
+        """
+
+        for bb in self._bb_list:
+            try:
+                self._thresholds[id(bb)] = bb._threshold
```

Comment:
I don't like accessing protected members (`bb._threshold` ) and my linter warns me, but ignoring them might hide other real issues. It is also a bit inconsistent.